services/order_fetcher.rb

```diff
@@ -0,0 +1,26 @@
+require 'net/http'
+require 'uri'
+require 'json'
+
+class OrderFetcher
+  def initialize(client_id, client_secret)
+    @auth_client = ThrivaAuthClient.new(client_id, client_secret)
+  end
+
+  def fetch_orders(start_date, end_date)
+    access_token = @auth_client.generate_access_token
+    base_url = "https://api.euw2.development.thriva.io/v1/orders/"
+    query_params = "?filter[startDate]=#{start_date}&filter[endDate]=#{end_date}"
```

Comment:
should we use our [custom thriva client](https://github.com/arjun-thriva/tmd-the-researcher-be/blob/main/services/thriva_client.rb) for doing call this?